### PR TITLE
Service 6.2.x with Add value for NU1507 warning string format item (#4533)

### PR DESCRIPTION
**Pull request by jeffkl**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/1543

Regression? Last working version:

## Description
Cherry picked 29af687868105f782edb6e21e1103780ec78f0e7 onto `release-6.2.x`
```
D:\nuget3>git fetch origin release-6.2.x
...

D:\nuget3>git reset origin/release-6.2.x --hard
HEAD is now at eae7622b6 change IsEscrowMode to true for 6.2.0 (#4542)

D:\nuget3>git cherry-pick 29af687868105f782edb6e21e1103780ec78f0e7
Auto-merging src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs
[jeffkl-cherry-pick-29af687868105f782edb6e21e1103780ec78f0e7 3e702d797] Add value for NU1507 warning string format item (#4533)
 Author: Tim Mulholland <tlmii@users.noreply.github.com>
 Date: Mon Apr 4 11:52:38 2022 -0700
 2 files changed, 2 insertions(+), 1 deletion(-)
```

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [x] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

**Review by nkolev92 — APPROVED** [review]

**Review by erdembayar — APPROVED** [review]

**Comment by jeffkl** [comment]
Waiting on other security fixes and approval from @aortiz-msft before I merge this

**Pull request merged**
